core/sysparam.c

```diff
@@ -719,97 +761,65 @@ sysparam_status_t sysparam_get_string(const char *key, char **destptr) {
     return SYSPARAM_OK;
 }
 
-sysparam_status_t sysparam_get_int(const char *key, int32_t *result) {
-    char *buffer;
-    char *endptr;
+sysparam_status_t sysparam_get_int32(const char *key, int32_t *result) {
     int32_t value;
+    size_t actual_length;
+    bool is_binary;
     sysparam_status_t status;
 
-    status = sysparam_get_string(key, &buffer);
+    status = sysparam_get_data_static(key, (uint8_t *)&value, sizeof(int32_t),
+                                      &actual_length, &is_binary);
     if (status != SYSPARAM_OK) return status;
-    value = strtol(buffer, &endptr, 0);
-    if (*endptr) {
-        // There was extra crap at the end of the string.
-        free(buffer);
+    if (!is_binary || actual_length != sizeof(int32_t))
         return SYSPARAM_PARSEFAILED;
-    }
-
     *result = value;
-    free(buffer);
-    return SYSPARAM_OK;
+    return status;
 }
 
-sysparam_status_t sysparam_get_bool(const char *key, bool *result) {
-    char *buffer;
+sysparam_status_t sysparam_get_int8(const char *key, int8_t *result) {
```

Comment:
I can see the use for a get_int8 but I would like to have the get_bool unchanged. As always, contributions improving functionality are very welcome without discussion but changes like this I think need to be discussed first.